pkg/storage/postgres/postgres.go

```diff
@@ -90,6 +92,11 @@ func (p *Postgres) Close() {
 	p.db.Close()
 }
 
+// UpdateCurrentTime allows overwriting current time
+func (p *Postgres) UpdateCurrentTime(timeNow storage.TimeProvider) {
+	p.currentTime = timeNow
```

Comment:
this poses a race condition

goroutineA is calling `Write` which calls `p.timeProvider.Now()`
goroutineB is calling `SetTimeProvider` which assignes `p.timeProvider = provider`

The race occurs on the read and assignment.

To protect ourselves from this we should take care to either synchronize concurrent access or use `sync.Once`.

e.g.

```
var once sync.Once

func (p *Postgres) SetTimeProvider(provider storage.TimeProvider) {
    once.Do(func() {
        p.timeProvider = provider
    })
}
```

This approach seems like it'd be fine because it avoids the race and still allows you to have full control over the implementation/mock behavior of the `TimeProvider` at the caller. You'd be able to reset/fast-forward/rewind time with the mocked TimeProvider.

Reply:
I am a bit hesitant in adding synchronization (i.e., mutex/lock) to protect a method that should only be invoked in unit test.  Rather, we should make it explicit that the method SetTimeProvider must not be used outside .  I will add in check to ensure SetTimeProvider is a no-op in non-test code.

Reply:
As discussed offline, we will revert back to use the separate test for mysql and postgresql.  However, we will abstract out the Write functionality into common.